### ISSUE TITLE
Refactor Anomaly Tests to use General DataGenerators

### ISSUE THREAD

**Issue by cogmission** [numenta/nupic#1871]
https://github.com/numenta/nupic/blob/master/tests/unit/nupic/algorithms/anomaly_likelihood_test.py#L64  This code should be extracted to a generic generator and the above test should be refactored to use the general data generator classes.

**Comment by scottpurdy** [comment]
@cogmission - hopefully we don't have to create a new data generator and one of the existing ones will work

**Comment by cogmission** [comment]
Yep. As long as it's clear by the name what kind of data the class generates?
